src/main/java/org/springframework/integration/kafka/dsl/KafkaSubscribableChannelSpec.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.kafka.dsl;
+
+import org.springframework.integration.kafka.channel.AbstractKafkaChannel;
+import org.springframework.integration.kafka.channel.SubscribableKafkaChannel;
+import org.springframework.kafka.config.KafkaListenerContainerFactory;
+import org.springframework.kafka.core.KafkaTemplate;
+
+/**
+ * Spec for a subscribable channel.
+ *
+ * @author Gary Russell
+ * @since 3.3
+ *
+ */
+public class KafkaSubscribableChannelSpec extends AbstractKafkaChannelSpec<KafkaSubscribableChannelSpec> {
+
+	private final KafkaListenerContainerFactory<?> factory;
+
+	private final boolean pubSub;
+
+	protected KafkaSubscribableChannelSpec(KafkaTemplate<?, ?> template, KafkaListenerContainerFactory<?> factory,
+			String topic, boolean pubSub) {
+
+		super(template, topic);
+		this.factory = factory;
+		this.pubSub = pubSub;
+	}
+
+	@Override
+	protected AbstractKafkaChannel doGet() {
+		this.channel = new SubscribableKafkaChannel(this.template, this.factory, this.topic, this.pubSub);
```

Comment:
Looks like this could be created in the ctor and therefore it doesn't look like we would need to override this `doGet()` method.